### DISCUSSION TITLE
File must be less than 3072kb - Problem

### DISCUSSION THREAD

**Discussion by w3sta** [tweepy/tweepy#1505] [Questions] [answered]
Dear Github-Com.!

I got a question, I am currently programming a code from a friend and I want to upload photos from Discord to Twitter. Everytime when I am uploading a picture in Discord bigger than 3072kb it wont upload...

Code:
upload_tweet = api.update_with_media(filename, status=tweet_text) 
tweet_id = str(upload_tweet.id)

I think here is the mistake, but I can't find it.

Warning:
Traceback (most recent call last):
  File "/home/container/.local/lib/python3.9/site-packages/discord/client.py", line 333, in _run_event
    await coro(*args, **kwargs)
  File "/home/container/main.py", line 49, in on_message
    upload_tweet = api.update_with_media(filename, status=tweet_text)
  File "/home/container/.local/lib/python3.9/site-packages/tweepy/api.py", line 279, in update_with_media
    headers, post_data = API._pack_image(filename, 3072,
  File "/home/container/.local/lib/python3.9/site-packages/tweepy/api.py", line 1398, in _pack_image
    raise TweepError('File is too big, must be less than %skb.'
tweepy.error.TweepError: File is too big, must be less than 3072kb.

**Comment by Harmon758** [comment, marked as answer]
The [POST statuses/update_with_media endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/post-and-engage/api-reference/post-statuses-update_with_media) that [`API.update_with_media`](https://tweepy.readthedocs.io/en/latest/api.html#API.update_with_media) uses has been deprecated, but its documented image file size limit is 3 MB.

This check prior to making the request to the endpoint will probably be removed in #1486, but the endpoint will just error instead when you pass a file that's too large.

Instead, you should use the [POST media/upload endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/api-reference/post-media-upload) with [`API.media_upload`](https://tweepy.readthedocs.io/en/latest/api.html#API.media_upload) and pass `media_ids` to the [POST statuses/update endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/post-and-engage/api-reference/post-statuses-update) with [`API.update_status`](https://tweepy.readthedocs.io/en/latest/api.html#API.update_status).

For code block usage, see https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks.

**Reply by w3sta** [reply]
Hey, thank you for helping me.

I am definitelly not a pro in what i am doing and now I am not getting an error, but when I post the picture now in Discord the correct text comes, but the tweet won't show up on Twitter

```
   upload_tweet = api.media_upload(filename, status=tweet_text) 
   media_ids = str(api.update_status)
```

**Reply by Harmon758** [reply]
[`API.update_status`](https://tweepy.readthedocs.io/en/latest/api.html#API.update_status) is a method, and [`API.media_upload`](https://tweepy.readthedocs.io/en/latest/api.html#API.media_upload) doesn't have a `status` parameter.
See the linked documentation.
